Wikipedia/UI-V5/WMFNearbyViewModel.m

```diff
@@ -0,0 +1,193 @@
+//
+//  WMFNearbyViewModel.m
+//  Wikipedia
+//
+//  Created by Brian Gerstle on 9/9/15.
+//  Copyright (c) 2015 Wikimedia Foundation. All rights reserved.
+//
+
+#import "WMFNearbyViewModel.h"
+
+// Location & Fetching
+#import "WMFLocationManager.h"
+#import "WMFLocationSearchFetcher.h"
+
+// Models
+#import "WMFLocationSearchResults.h"
+#import "MWKLocationSearchResult.h"
+#import "MWKHistoryEntry.h"
+#import "CLLocation+WMFApproximateEquality.h"
+#import "CLLocation+WMFBearing.h"
+
+// Frameworks
+#import "Wikipedia-Swift.h"
+#import "PromiseKit.h"
+
+@interface WMFNearbyViewModel ()
+<WMFNearbyControllerDelegate>
+
+@property (nonatomic, strong) WMFLocationSearchFetcher* locationSearchFetcher;
+@property (nonatomic, strong) WMFLocationManager* locationManager;
+@property (nonatomic, assign) NSUInteger resultLimit;
+
+@property (nonatomic, strong, readwrite) WMFLocationSearchResults* locationSearchResults;
+
+@property (nonatomic, strong) NSMutableIndexSet* autoUpdatingIndexes;
+
+@end
+
+@implementation WMFNearbyViewModel
+
+- (instancetype)initWithSite:(MWKSite*)site
+                 resultLimit:(NSUInteger)resultLimit
+             locationManager:(WMFLocationManager* __nullable)locationManager {
+    return [self initWithSite:site
+                  resultLimit:resultLimit
+              locationManager:locationManager ? : [[WMFLocationManager alloc] init]
+                      fetcher:[[WMFLocationSearchFetcher alloc] init]];
+}
+
+- (instancetype)initWithSite:(MWKSite*)site
+                 resultLimit:(NSUInteger)resultLimit
+             locationManager:(WMFLocationManager*)locationManager
+                     fetcher:(WMFLocationSearchFetcher*)locationSearchFetcher {
+    self = [super init];
+    if (self) {
+        self.autoUpdatingIndexes   = [NSMutableIndexSet indexSet];
+        self.site                  = site;
+        self.resultLimit           = resultLimit;
+        self.locationSearchFetcher = locationSearchFetcher;
+        // !!!: Must setup location manager last to prevent delegate callbacks firing before fetcher is setup
+        self.locationManager          = locationManager;
+        self.locationManager.delegate = self;
+    }
+    return self;
+}
+
+- (void)dealloc {
+    [self stopUpdates];
+}
+
+- (void)setSite:(MWKSite* __nonnull)site {
```

Comment:
It does, but I actually don't think we'd propagate search site changes here (since nobody has a direct reference to the article list controller that gets pushed)